diff --git a/archive_and_clean.sh b/archive_and_clean.sh
index 07b02f5..1c6f897 100755
--- a/archive_and_clean.sh
+++ b/archive_and_clean.sh
@@ -30,7 +30,7 @@ done
 
 #archive and compress previous day's gnss data.
 #find . -maxdepth 1 -type f -mtime -1 -mmin +60 -name "*.ubx*" -exec tar -jcvf ${archive_name} --remove-files {} +;
-find . -maxdepth 1 -type f -mtime -960 -mmin +60 \( -name "*.rtcm*" -o -name "*.nov*" -o -name "*.oem*" -o -name "*.ubx*" -o -name "*.ss2*" -o -name "*.hemis*" -o -name "*.stq*" -o -name "*.javad*" -o -name "*.nvs*" -o -name "*.binex*" -o -name "*.sbf*" \) -exec zip -m9 ${archive_name} {} +;
+find . -maxdepth 1 -type f -mtime -960 -mmin +60 \( -name "*.rtcm*" -o -name "*.nov*" -o -name "*.oem*" -o -name "*.ubx*" -o -name "*.ss2*" -o -name "*.hemis*" -o -name "*.stq*" -o -name "*.javad*" -o -name "*.nvs*" -o -name "*.binex*" -o -name "*.sbf*" \) -exec ${BASEDIR}/tools/create_archive.sh {} +;
 
 #delete gnss data older than x days.
 #find . -maxdepth 1 -type f -name "*.tar.bz2" -mtime +${archive_rotate} -delete
