diff --git a/tools/convbin.sh b/tools/convbin.sh
index 91a8020..07fc01d 100755
--- a/tools/convbin.sh
+++ b/tools/convbin.sh
@@ -5,18 +5,54 @@ SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 source <( grep '=' "${SCRIPT_DIR}"/../settings.conf )
 RAW_ARCHIVE=$1
 DATA_DIR=$2
-MOUNT_NAME=$mnt_name_a
+MARKER_NAME="${marker_name}"
+if [[ -z "${MARKER_NAME}" ]]; then
+   MARKER_NAME="${mnt_name_a}"
+fi
+MARKER_NUMBER=`echo "${marker_number}" | awk -F ',' '{print $1}'`
 RAW_TYPE=$receiver_format
 RINEX_TYPE=$3
+FILENAME_MODE=$rinex_filename
 CONVBIN_PATH=$(type -P convbin)
-ANT_POSITION=$(echo "${position}" | cs2cs  EPSG:4979 EPSG:4978 -f "%.2f" | sed 's/\s/\//g')
-RECEIVER="${receiver}"
+if [[ "${position}" == "0.00 0.00 0.00" ]]; then
+   ANT_POSITION=
+   MARKER_TYPE=NON_GEODETIC
+else
+   ANT_POSITION=$(echo "${position}" | cs2cs  EPSG:4979 EPSG:4978 -f "%.2f" | sed 's/\s/\//g')
+   MARKER_TYPE=GEODETIC
+fi
+RECEIVER=$(echo "${receiver}" | sed  's/Septentrio_mosaic-/SEPT MOSAIC-/')
 REC_VERSION="${receiver_firmware}"
+REC_SERIAL="${receiver_serial_number}"
 REC_OPTION=''
 [[ -n $ntrip_a_receiver_options ]] && REC_OPTION="${ntrip_a_receiver_options}"
-ANT_TYPE="${antenna_info}"
 RTKBASE_VERSION='RTKBase v'"${version}"
 
+ANTNAME=`echo "${antenna_info}" | awk -F ',' '{print $1}'`
+ANTSERIAL=`echo "${antenna_info}" | awk -F ',' '{print $2}'`
+if [[ "${ANTNAME}" == "ELT0123" ]] || [[ "${ANTNAME}" == "ELT0323" ]]; then
+   ANTNAME="HXCSX627A       NONE"
+fi
+if [[ "${ANTSERIAL}" == "" ]]; then
+   ANTSERIAL=0000
+fi
+ANT_INFO="${ANTSERIAL}/${ANTNAME}"
+#echo ANT_INFO=\"${ANT_INFO}\"
+
+ANT_H=`echo "${antenna_info}" | awk -F ',' '{print $4}'`
+ANT_E=`echo "${antenna_info}" | awk -F ',' '{print $5}'`
+ANT_N=`echo "${antenna_info}" | awk -F ',' '{print $6}'`
+if [[ -z "${ANT_H}" ]]; then ANT_H="0.0000"; fi
+if [[ -z "${ANT_E}" ]]; then ANT_E="0.0000"; fi
+if [[ -z "${ANT_N}" ]]; then ANT_N="0.0000"; fi
+ANT_DELTA="${ANT_H}/${ANT_E}/${ANT_N}"
+#echo ANT_DELTA=\"${ANT_DELTA}\"
+
+OBSERVER=`echo "${rinex_observer}" | awk -F ',' '{print $1}'`
+AGENCY=`echo "${rinex_observer}" | awk -F ',' '{print $2}'`
+AGENT_INFO="${OBSERVER}/${AGENCY}"
+#echo AGENT_INFO=\"${AGENT_INFO}\"
+
 extract_raw_file() {
   raw_file=$(unzip -l "${RAW_ARCHIVE}" "*.${RAW_TYPE}" | awk '/-----/ {p = ++p % 2; next} p {print $NF}')
   test $(echo "${raw_file}" | wc -l) -gt 1 && echo 'Error: There is more than 1 file in this archive' 1>&2 && exit 1
@@ -29,9 +65,9 @@ extract_raw_file() {
 convert_to_rinex_ign() {
   echo "- CREATING RINEX " "${RINEX_FILE}"
   "${CONVBIN_PATH}" "${raw_file}" -v 2.11 -r "${RAW_TYPE}"       \
-        -hc "${RTKBASE_VERSION}" -hm "${MOUNT_NAME}"    \
-        -hp "${ANT_POSITION}" -ha 0000/"${ANT_TYPE}"   \
-        -hr 0000/"${RECEIVER}"/"${REC_VERSION}"        \
+        -hc "${RTKBASE_VERSION}" -hm "${MARKER_NAME}" -hn "${MARKER_NUMBER}" -ht "${MARKER_TYPE}" \
+        -hp "${ANT_POSITION}" -ha "${ANT_INFO}" -ho "${AGENT_INFO}" -hd "${ANT_DELTA}" \
+        -hr "${REC_SERIAL}"/"${RECEIVER}"/"${REC_VERSION}"        \
         -f 2 -y R -y E -y J -y S -y C -y I             \
         -od -os -oi -ot -ti 30 -tt 0.005                   \
         -ro "${REC_OPTION}" -o "${RINEX_FILE}"
@@ -40,7 +76,7 @@ convert_to_rinex_ign() {
 
 convert_to_rinex_ign_bis() {
   echo "- CREATING RINEX " "${RINEX_FILE}"
-  args="${raw_file}"' -v 2.11 -r '"${RAW_TYPE}"' -hc '"${RTKBASE_VERSION}"' -hm '"${MOUNT_NAME}"' -hp '"${ANT_POSITION}"' -ha 0000/'"${ANT_TYPE}"' -hr 0000/'"${RECEIVER}"'/'"${REC_VERSION}"' -f 2 -y R -y E -y J -y S -y C -y I -od -os -oi -ot -ti 30 -tt 0 -ro '"${REC_OPTION}"' -o '"${RINEX_FILE}"
+  args="${raw_file}"' -v 2.11 -r '"${RAW_TYPE}"' -hc '"${RTKBASE_VERSION}"' -hm '"${MARKER_NAME}"' -hn '"${MARKER_NUMBER}"' -ht '"${MARKER_TYPE}"' -hp '"${ANT_POSITION}"' -ha '"${ANT_INFO}"' -ho '"${AGENT_INFO}"' -hd '"${ANT_DELTA}"' -hr '"${REC_SERIAL}"'/'"${RECEIVER}"'/'"${REC_VERSION}"' -f 2 -y R -y E -y J -y S -y C -y I -od -os -oi -ot -ti 30 -tt 0 -ro '"${REC_OPTION}"' -o '"${RINEX_FILE}"
   "${CONVBIN_PATH}" "${args}"
     return $?
 }
@@ -48,9 +84,9 @@ convert_to_rinex_ign_bis() {
 convert_to_rinex_nrcan() {
   echo "- CREATING RINEX " "${RINEX_FILE}"
   "${CONVBIN_PATH}" "${raw_file}" -v 3.04 -r "${RAW_TYPE}"       \
-        -hc "${RTKBASE_VERSION}" -hm "${MOUNT_NAME}"   \
-        -hp "${ANT_POSITION}" -ha 0000/"${ANT_TYPE}"   \
-        -hr 0000/"${RECEIVER}"/"${REC_VERSION}"        \
+        -hc "${RTKBASE_VERSION}" -hm "${MARKER_NAME}" -hn "${MARKER_NUMBER}" -ht "${MARKER_TYPE}"  \
+        -hp "${ANT_POSITION}" -ha "${ANT_INFO}" -ho "${AGENT_INFO}" -hd "${ANT_DELTA}" \
+        -hr "${REC_SERIAL}"/"${RECEIVER}"/"${REC_VERSION}"        \
         -f 2 -y J -y S -y C -y I                  \
         -od -os -oi -ot -ti 30 -tt 0                   \
         -ro "${REC_OPTION}" -o "${RINEX_FILE}"
@@ -61,9 +97,9 @@ convert_to_rinex_nrcan() {
 convert_to_rinex_30s_full() {
   echo "- CREATING RINEX " "${RINEX_FILE}"
   "${CONVBIN_PATH}" "${raw_file}" -v 3.04 -r "${RAW_TYPE}"       \
-        -hc "${RTKBASE_VERSION}" -hm "${MOUNT_NAME}"   \
-        -hp "${ANT_POSITION}" -ha 0000/"${ANT_TYPE}"   \
-        -hr 0000/"${RECEIVER}"/"${REC_VERSION}"        \
+        -hc "${RTKBASE_VERSION}" -hm "${MARKER_NAME}"  -hn "${MARKER_NUMBER}" -ht "${MARKER_TYPE}"  \
+        -hp "${ANT_POSITION}" -ha "${ANT_INFO}" -ho "${AGENT_INFO}" -hd "${ANT_DELTA}" \
+        -hr "${REC_SERIAL}"/"${RECEIVER}"/"${REC_VERSION}"        \
         -od -os -oi -ot -ti 30 -tt 0                   \
         -ro "${REC_OPTION}" -o "${RINEX_FILE}"
     return $?
@@ -73,9 +109,9 @@ convert_to_rinex_30s_full() {
 convert_to_rinex_1s_full() {
   echo "- CREATING RINEX " "${RINEX_FILE}"
   "${CONVBIN_PATH}" "${raw_file}" -v 3.04 -r "${RAW_TYPE}"       \
-        -hc "${RTKBASE_VERSION}" -hm "${MOUNT_NAME}"   \
-        -hp "${ANT_POSITION}" -ha 0000/"${ANT_TYPE}"   \
-        -hr 0000/"${RECEIVER}"/"${REC_VERSION}"        \
+        -hc "${RTKBASE_VERSION}" -hm "${MARKER_NAME}" -hn "${MARKER_NUMBER}" -ht "${MARKER_TYPE}"  \
+        -hp "${ANT_POSITION}" -ha "${ANT_INFO}" -ho "${AGENT_INFO}" -hd "${ANT_DELTA}" \
+        -hr "${REC_SERIAL}"/"${RECEIVER}"/"${REC_VERSION}"        \
         -od -os -oi -ot -ti 1 -tt 0                    \
         -ro "${REC_OPTION}" -o "${RINEX_FILE}"
     return $?
@@ -88,15 +124,84 @@ filedate=$(echo ${1:0:10})
 year2=$(echo ${1:2:2})
 
 test -z "${CONVBIN_PATH}" && echo 'Error: Convbin not found' && exit 1
-if [[ ${RINEX_TYPE} == 'ign' ]] ; then rnx_conversion_func='convert_to_rinex_ign' ; RINEX_FILE=$(echo "${filedate}"-"${MOUNT_NAME}"_"${RINEX_TYPE}"."${year2}"o)
-elif [[ ${RINEX_TYPE} == 'nrcan' ]] ; then rnx_conversion_func='convert_to_rinex_nrcan' ; RINEX_FILE=$(echo "${filedate}"-"${MOUNT_NAME}"_"${RINEX_TYPE}".obs)
-elif [[ ${RINEX_TYPE} == '30s_full' ]] ; then rnx_conversion_func='convert_to_rinex_30s_full' ; RINEX_FILE=$(echo "${filedate}"-"${MOUNT_NAME}"_"${RINEX_TYPE}".obs)
-elif [[ ${RINEX_TYPE} == '1s_full' ]] ; then rnx_conversion_func='convert_to_rinex_1s_full' ; RINEX_FILE=$(echo "${filedate}"-"${MOUNT_NAME}"_"${RINEX_TYPE}".obs)
+if [[ ${RINEX_TYPE} == 'ign' ]] ; then rnx_conversion_func='convert_to_rinex_ign' ; RINEX_FILE=$(echo "${filedate}"-"${MARKER_NAME}"_"${RINEX_TYPE}"."${year2}"o)
+elif [[ ${RINEX_TYPE} == 'nrcan' ]] ; then rnx_conversion_func='convert_to_rinex_nrcan' ; RINEX_FILE=$(echo "${filedate}"-"${MARKER_NAME}"_"${RINEX_TYPE}".obs)
+elif [[ ${RINEX_TYPE} == '30s_full' ]] ; then rnx_conversion_func='convert_to_rinex_30s_full' ; RINEX_FILE=$(echo "${filedate}"-"${MARKER_NAME}"_"${RINEX_TYPE}".obs)
+elif [[ ${RINEX_TYPE} == '1s_full' ]] ; then rnx_conversion_func='convert_to_rinex_1s_full' ; RINEX_FILE=$(echo "${filedate}"-"${MARKER_NAME}"_"${RINEX_TYPE}".obs)
+else echo Error: invalid rinex type \"${RINEX_TYPE}\"; exit 1
+fi
+
+marker4="${MARKER_NAME}____"
+marker4="${marker4:0:4}"
+day=$(date -d "${RAW_ARCHIVE:0:10}" +%j)
+markernum="${MARKER_NUMBER}0"
+markernum="${markernum:0:1}"
+file_extension="${RAW_ARCHIVE##*.}"
+
+iso3() {
+    local code2="$(iw reg get | sed -n "0,/country/s/^country \(.\+\):.*$/\1/p")"
+    if [[ "${code2}" == "00" ]]; then
+       code2=$(curl -s "http://ip-api.com/json/?fields=countryCode"  | jq -r '.countryCode')
+       #echo code2=${code2}
+    fi
+    if [[ "${#code2}" == "2" ]]; then
+       local code3="$(jq -r --arg code "${code2}" '."3166-1"[] | select(.alpha_2 == $code) | .alpha_3' /usr/share/iso-codes/json/iso_3166-1.json)"
+       local lastcode=$?
+       #echo code3=${code3} lastcode=${lastcode}
+       if [[ "${#code3}" == "3" ]] && [[ "${lastcode}" == "0" ]]; then
+          echo ${code3}
+       else
+          echo ${code2}_
+       fi
+    else
+       echo "UNK"
+    fi
+}
+
+num2letter() {
+    printf "\\$(printf '%03o' $((97 + 10#$1)))"
+}
+
+getHHMM() {
+   local hh="00"
+   local mm="00"
+   if [[ "$1" =~ _([0-9]{2})(-([0-9]{2}))? ]]; then
+      hh=${BASH_REMATCH[1]}
+      mm=${BASH_REMATCH[3]:-00}
+   fi
+   echo "${hh}${mm}"
+}
+
+if [[ ${FILENAME_MODE} == '1' ]]; then
+   if [[ $file_extension == 'zip' ]]; then
+      hourletter='0'
+   else
+      hhmm=$(getHHMM "${RAW_ARCHIVE}")
+      hourletter="$(num2letter ${hhmm:0:2})"
+   fi
+   #echo hhmm=${hhmm} hourletter=${hourletter}
+   RINEX_FILE=$(echo "${marker4}""${day}"${hourletter}."${year2}"o)
+elif [[ ${FILENAME_MODE} == '2' ]]; then
+   year4=$(echo ${RAW_ARCHIVE:0:4})
+   hhmm=$(getHHMM "${RAW_ARCHIVE}")
+   if [[ ${RINEX_TYPE} == '30s_full' ]]; then
+      datafreq="30S"
+   else
+      datafreq="01S"
+   fi
+   MONUMENT_INDEX=`echo "${marker_number}" | awk -F ',' '{print $2}'`
+   RECEIVER_INDEX=`echo "${marker_number}" | awk -F ',' '{print $3}'`
+   if [[ -z "${MONUMENT_INDEX}" ]]; then
+      MONUMENT_INDEX=0
+   fi
+   if [[ -z "${RECEIVER_INDEX}" ]]; then
+      RECEIVER_INDEX=0
+   fi
+   RINEX_FILE=$(echo "${marker4}""${MONUMENT_INDEX}""${RECEIVER_INDEX}""$(iso3)"_R_"${year4}""${day}""${hhmm}"_01D_"${datafreq}"_MO.rnx.obs)
 fi
 
 #Let's launch the CONVBIN process
 echo "- Processing on	""${RAW_ARCHIVE}"
-file_extension="${RAW_ARCHIVE##*.}"
 if [[ $file_extension == 'zip' ]]
   then
     extract_raw_file
@@ -105,6 +210,10 @@ if [[ $file_extension == 'zip' ]]
 fi
 ${rnx_conversion_func}
 return_code=$?
+if [[ ! -f "${RINEX_FILE}" ]]; then
+   echo RINEX not created
+   exit 1
+fi
 echo -n 'rinex_file='"${RINEX_FILE}"
 [[ $file_extension == 'zip' ]] && rm "${raw_file}"
 exit $return_code
\ No newline at end of file
