diff --git a/web_app/LogManager.py b/web_app/LogManager.py
index 1371110..3128639 100644
--- a/web_app/LogManager.py
+++ b/web_app/LogManager.py
@@ -29,7 +29,7 @@ from log_converter import convbin
 
 class LogManager():
 
-    supported_solution_formats = ["llh", "xyz", "enu", "nmea", "erb", "zip", "bz2", "tar", "tag"]
+    supported_solution_formats = ["llh", "xyz", "enu", "nmea", "erb", "zip", "bz2", "tar", "tag", "log"]
 
     def __init__(self, rtklib_path, log_path):
 
