diff --git a/web_app/network_infos.py b/web_app/network_infos.py
index f71b92e..90b36f1 100644
--- a/web_app/network_infos.py
+++ b/web_app/network_infos.py
@@ -44,7 +44,7 @@ def get_up_if():
     if_stats.pop('lo')
     log.debug(if_stats)
     if_stats = {k:v for (k,v) in if_stats.items() if if_stats[k].isup} # keep only up interface
-    if_stats = {k:v for (k,v) in if_stats.items() if if_stats[k].speed > 0 or 'pointopoint' in if_stats[k].flags} # keep only interface with speed > 0
+    if_stats = {k:v for (k,v) in if_stats.items() if if_stats[k].speed > 0 or 'pointopoint' in if_stats[k].flags or k.startswith('wlan')} # keep only interface with speed > 0
     if_stats = {k:v for (k,v) in if_stats.items() if not k.startswith('docker')} # remove docker interface
     return if_stats
 
@@ -87,7 +87,8 @@ def get_interfaces_infos():
         if conn_name:
             device_info["conn_name"] = conn_name
         device_info["hwaddr"] = nmcli.device.show(k).get('GENERAL.HWADDR')
-        interfaces_infos.append(device_info)
+        if (len(ipv4) > 0) or (len(ipv6) > 0):
+            interfaces_infos.append(device_info)
     return interfaces_infos
 
 def arg_parse():
