diff --git a/web_app/RTKBaseConfigManager.py b/web_app/RTKBaseConfigManager.py
index eabd303..a5c7211 100644
--- a/web_app/RTKBaseConfigManager.py
+++ b/web_app/RTKBaseConfigManager.py
@@ -110,7 +110,7 @@ class RTKBaseConfigManager:
             and remove the single quotes.      
         """
         ordered_main = [{"source_section" : "main"}]
-        for key in ("position", "com_port", "com_port_settings", "receiver", "receiver_firmware", "receiver_format", "antenna_info", "tcp_port", "gnss_rcv_web_ip", "gnss_rcv_web_proxy_port"):
+        for key in ("position", "com_port", "com_port_settings", "receiver", "receiver_firmware", "receiver_format", "antenna_info", "tcp_port", "gnss_rcv_web_ip", "gnss_rcv_web_proxy_port", "mobile_modem_web_ip", "mobile_modem_web_proxy_port"):
             ordered_main.append({key : self.config.get('main', key).strip("'")})
         return ordered_main
 
