diff --git a/web_app/server.py b/web_app/server.py
index e36bbc4..6544aa7
--- a/web_app/server.py
+++ b/web_app/server.py
@@ -99,9 +99,12 @@ rtk = RTKLIB(socketio,
             log_path=app.config["DOWNLOAD_FOLDER"],
             )
 
-services_list = [{"service_unit" : "str2str_tcp.service", "name" : "main"},
-                 {"service_unit" : "str2str_ntrip_A.service", "name" : "ntrip_A"},
-                 {"service_unit" : "str2str_ntrip_B.service", "name" : "ntrip_B"},
+services_list = [{"service_unit" : "str2str_tcp.service", "name" : "main", "flag" : True},
+                 {"service_unit" : "str2str_ntrip_A.service", "name" : "ntrip_A", "flag" : True},
+                 {"service_unit" : "str2str_ntrip_B.service", "name" : "ntrip_B", "flag" : True},
+                 {"service_unit" : "str2str_ntrip_C.service", "name" : "ntrip_C", "flag" : True},
+                 {"service_unit" : "str2str_ntrip_D.service", "name" : "ntrip_D", "flag" : True},
+                 {"service_unit" : "str2str_ntrip_E.service", "name" : "ntrip_E", "flag" : True},
                  {"service_unit" : "str2str_local_ntrip_caster.service", "name" : "local_ntrip_caster"},
                  {"service_unit" : "str2str_rtcm_svr.service", "name" : "rtcm_svr"},
                  {'service_unit' : 'str2str_rtcm_serial.service', "name" : "rtcm_serial"},
@@ -109,12 +112,14 @@ services_list = [{"service_unit" : "str2str_tcp.service", "name" : "main"},
                  {'service_unit' : 'rtkbase_archive.timer', "name" : "archive_timer"},
                  {'service_unit' : 'rtkbase_archive.service', "name" : "archive_service"},
                  {'service_unit' : 'rtkbase_raw2nmea.service', "name" : "raw2nmea"},
-                 {'service_unit' : 'rtkbase_gnss_web_proxy.service', "name": "RTKBase Reverse Proxy for Gnss receiver Web Server"}
+                 {'service_unit' : 'rtkbase_gnss_web_proxy.service', "name": "RTKBase Reverse Proxy for Gnss receiver Web Server"},
+                 {'service_unit' : 'rtkbase_modem_web_proxy.service', "name" : "RTKBase Reverse Proxy for Mobile Modem Web Server"},
                  ]
 
 #Delay before rtkrcv will stop if no user is on status.html page
-rtkcv_standby_delay = 600
+rtkcv_standby_delay = 129600
 connected_clients = 0
+rebooted_state = False
 
 class StandaloneApplication(gunicorn.app.base.BaseApplication):
     def __init__(self, app, options=None):
@@ -217,9 +222,21 @@ def repaint_services_button(services_list):
         result = exit-code or signal => red button (danger)
     """ 
     for service in services_list:
-        """ 
-        if service.get("status") == "running":
-            service["btn_color"] = "success"
+        isflag = None
+        if service.get("flag") == True:
+            path = os.path.join(rtkbase_path, service.get("name") + ".flg")
+            isflag = os.path.exists(path)
+            if isflag:
+               if service.get("status") != "dead":
+                  with open(path, 'rt') as f:
+                      service["error"] = f.read()
+        if service.get("status") == "start":
+            service["btn_color"] = "info"
+        elif service.get("status") == "running":
+            if isflag:
+                service["btn_color"] = "warning"
+            else:
+                service["btn_color"] = "success"
             service["btn_off_color"] = "outline-secondary"
         elif service.get("result") == "exit-code" or service.get("result") == "signal":
             service["btn_color"] = "warning"
@@ -233,9 +250,8 @@ def repaint_services_button(services_list):
         if service.get("state_ok") == False:
             service["btn_off_color"] = "outline-danger"
         elif service.get("state_ok") == True:
-            service["btn_off_color"] = "outline-secondary" 
+            service["btn_off_color"] = "outline-secondary"
         """
-
         if service.get("status") == "running":
             service["btn_color"] = "success"
         elif service.get("status") == "auto-restart":
@@ -244,6 +260,10 @@ def repaint_services_button(services_list):
             service["btn_off_color"] = "outline-secondary"
         elif service.get("result") == "exit-code" or service.get("result") == "signal":
             service["btn_off_color"] = "danger"
+        """
+
+        #if service.get("flag") == True:
+        #   print(service.get("name"), service.get("status"), service.get("result"), service.get("state_ok"))
 
     return services_list
 
@@ -407,13 +427,111 @@ def download_update(update_path):
     else:
         return update_archive
 
+@socketio.on("check update ELT", namespace="/test")
+def check_updateELT(json_msg):
+    branch = json_msg.get("branch")
+    if branch is None:
+       branch = "main"
+    print("Check update ELT started. branch:", branch)
+    new_release = {}
+    source_url = "https://raw.githubusercontent.com/GNSSOEM/ELT_RTKBase/refs/heads/" + branch + "/Description.json"
+    version_txt = os.path.join(rtkbase_path, "..", "version.txt")
+    try:
+        with open(version_txt, 'rt') as f:
+            current_release = int(f.read())
+        response = requests.get(source_url)
+        if response.ok:
+           response_json = response.json()
+           latest_release = int(response_json.get("version"))
+           #print("response_json=", response_json)
+           #print("current_release=", current_release)
+           #print("latest_release=", latest_release)
+           if latest_release > current_release:
+               new_release = response_json
+        else:
+           new_release = { "error" : "Description.json not downloded, error " + str(response.status_code)}
+    except Exception as e:
+        print("Check update ELT error: ", e)
+        new_release = { "error" : repr(e)}
+
+    #print("new_release=", new_release)
+    socketio.emit("new release", json.dumps(new_release), namespace="/test")
+    return new_release
+
+@socketio.on("update rtkbase ELT", namespace="/test")
+def update_rtkbaseELT(json_msg):
+    """
+        Check if a ELT RTKBase update exists, download it and update rtkbase
+        if update_file is a link to a file, use it to update rtkbase (mainly used for dev purpose)
+    """
+    update_file = json_msg.get("update_file")
+    branch = json_msg.get("branch")
+    if branch is None:
+       branch = "main"
+    print("update rtkbase ELT started. branch:", branch, "update_file:", update_file)
+
+    update_url = "https://raw.githubusercontent.com/GNSSOEM/ELT_RTKBase/refs/heads/" + branch + "/install.sh"
+    update_archive = os.path.join(rtkbase_path, "..", "update", "update.sh")
+    update_dir = os.path.join(rtkbase_path, "..", "update")
+
+    if update_file is None:
+        #Download update
+        #update_archive = download_update(update_url)
+        try:
+            response = requests.get(update_url)
+            if response.ok:
+                with open(update_archive, "wb") as f:
+                    f.write(response.content)
+                print("update downloaded in",update_archive)
+            else:
+                error = update_url + " not downloded, error " + str(response.status_code)
+                socketio.emit("downloading_update", json.dumps({"result": 'false', "error" : error}), namespace="/test")
+                return
+        except Exception as e:
+            error = "Error: Can't download update - " + repr(e)
+            print(error)
+            socketio.emit("downloading_update", json.dumps({"result": 'false', "error" : error}), namespace="/test")
+            return
+    else:
+        #update from file
+        update_file.save(update_archive)
+        print("update stored in",update_archive)
+
+    os.chmod(update_archive,0o755)
+    socketio.emit("downloading_update", json.dumps({"result": 'true'}), namespace="/test")
+
+    answer = subprocess.run([update_archive, "-s"], encoding="UTF-8", stderr=subprocess.PIPE, stdout=subprocess.PIPE, check=False)
+    if answer.returncode != 0:
+        if answer.stderr != "":
+            error = answer.stderr
+        else:
+            error = answer.stdout
+        #print("update -s answer.returncode = ", answer.returncode)
+        #print("update -s error:", error)
+        if answer.returncode != 254:
+           result = "Error !"
+        else:
+           result = "Status"
+        socketio.emit("updating_rtkbase_stopped", json.dumps({"result": result, "error" : error.splitlines()}), namespace="/test")
+        print("Checking OS release failed. Update aborted!")
+    else : #if ok, launch update script
+        exec_update=os.path.join(rtkbase_path, "tools", "exec_update.sh")
+        print("Launch update", exec_update)
+        socketio.emit("updating_rtkbase", namespace="/test")
+        rtk.shutdownBase()
+        time.sleep(1)
+        #update_service=ServiceController('rtkbase_update.service')
+        #update_service.start()
+        subprocess.Popen([exec_update, "-U"], cwd=update_dir)
+
 @app.before_request
 def inject_global_infos():
     """
         Insert various informations as global variables for Flask/Jinja
     """
     g.version = rtkbaseconfig.get("general", "version")
-    g.station_name = rtkbaseconfig.get_ntrip_A_settings()[4]['mnt_name_A']
+    g.station_name = rtkbaseconfig.get_ntrip_A_settings()[6]['mnt_name_A']
+    g.elt_version = rtkbaseconfig.get("general", "elt_version")
     g.sbc_model = get_sbc_model()
 
 @login.user_loader
@@ -442,7 +560,7 @@ def settings_page():
     if request.method == 'POST':
         uploaded_file = request.files['file']
         if uploaded_file.filename != '':
-            update_rtkbase(uploaded_file)
+            update_rtkbaseELT({"update_file": uploaded_file})
         else:
             print("wrong update file")
         return ('', 204)
@@ -453,11 +571,20 @@ def settings_page():
                                             path=host_url.path,
                                             params=host_url.params, query=host_url.query,
                                             fragment=host_url.fragment)
+    mobile_modem_url = urllib.parse.ParseResult(scheme=host_url.scheme,
+                                            netloc="{}:{}".format(host_url.hostname, rtkbaseconfig.get("main", "mobile_modem_web_proxy_port")),
+                                            path=host_url.path,
+                                            params=host_url.params, query=host_url.query,
+                                            fragment=host_url.fragment)
     #TODO use dict and not list
     main_settings = rtkbaseconfig.get_main_settings()
     main_settings.append(gnss_rcv_url.geturl())
+    main_settings.append(mobile_modem_url.geturl())
     ntrip_A_settings = rtkbaseconfig.get_ntrip_A_settings()
     ntrip_B_settings = rtkbaseconfig.get_ntrip_B_settings()
+    ntrip_C_settings = rtkbaseconfig.get_ntrip_C_settings()
+    ntrip_D_settings = rtkbaseconfig.get_ntrip_D_settings()
+    ntrip_E_settings = rtkbaseconfig.get_ntrip_E_settings()
     local_ntripc_settings = rtkbaseconfig.get_local_ntripc_settings()
     rtcm_svr_settings = rtkbaseconfig.get_rtcm_svr_settings()
     rtcm_client_settings = rtkbaseconfig.get_rtcm_client_settings()
@@ -465,10 +592,24 @@ def settings_page():
     rtcm_udp_client_settings = rtkbaseconfig.get_rtcm_udp_client_settings()
     rtcm_serial_settings = rtkbaseconfig.get_rtcm_serial_settings()
     file_settings = rtkbaseconfig.get_file_settings()
+    have_mobile = os.path.exists("/sys/class/net/mobile")
+
+    #start or stop  rtkbase_modem_web_proxy.service
+    for service in services_list:
+        if service["service_unit"] == 'rtkbase_modem_web_proxy.service':
+           service["active"] = service["unit"].isActive()
+           if service["active"] != have_mobile:
+              if have_mobile:
+                 service["unit"].start()
+              else:
+                 service["unit"].stop()
 
     return render_template("settings.html", main_settings = main_settings,
                                             ntrip_A_settings = ntrip_A_settings,
                                             ntrip_B_settings = ntrip_B_settings,
+                                            ntrip_C_settings = ntrip_C_settings,
+                                            ntrip_D_settings = ntrip_D_settings,
+                                            ntrip_E_settings = ntrip_E_settings,
                                             local_ntripc_settings = local_ntripc_settings,
                                             rtcm_svr_settings = rtcm_svr_settings,
                                             rtcm_client_settings = rtcm_client_settings,
@@ -476,7 +617,8 @@ def settings_page():
                                             rtcm_udp_client_settings = rtcm_udp_client_settings,
                                             rtcm_serial_settings = rtcm_serial_settings,
                                             file_settings = file_settings,
-                                            os_infos = distro.info(),)
+                                            os_infos = distro.info(),
+                                            have_mobile = have_mobile,)
 
 @app.route('/logs')
 @login_required
@@ -528,13 +670,22 @@ def diagnostic():
     """
     getServicesStatus()
     rtkbase_web_service = {'service_unit' : 'rtkbase_web.service', 'name' : 'RTKBase Web Server', 'active' : True}
+    rtkbase_check_internet_service = {'service_unit' : 'rtkbase_check_internet.service', 'name' : 'RTKBase show internet state on led'}
+    rtkbase_check_satelites_service = {'service_unit' : 'rtkbase_check_satelites.service', 'name' : 'RTKBase show internet state on led'}
+    RtkbaseSystemConfigure_service = {'service_unit' : 'RtkbaseSystemConfigure.service', 'name' : 'RTKBase system configure at start'}
+    rtkbase_septentrio_NAT_service = {'service_unit' : 'rtkbase_septentrio_NAT.service', 'name' : 'RTKBase configure Septentrio NAT at start'}
+    rtkbase_DHCP_service = {'service_unit' : 'rtkbase_DHCP.service', 'name' : 'RTKBase DHCP server for Septentrio'}
+    NetworkManager_service = {'service_unit' : 'NetworkManager.service', 'name' : 'Network Manager'}
+    gpsd_service = {'service_unit' : 'gpsd.service', 'name' : 'GPS (Global Positioning System) Daemon'}
+    chrony_service = {'service_unit' : 'chrony.service', 'name' : 'chrony, an NTP client/server'}
+
     logs = []
-    for service in services_list + [rtkbase_web_service]:
+    for service in services_list + [rtkbase_web_service, rtkbase_check_internet_service, rtkbase_check_satelites_service, RtkbaseSystemConfigure_service, rtkbase_septentrio_NAT_service, rtkbase_DHCP_service, NetworkManager_service, gpsd_service, chrony_service]:
         sysctl_status = subprocess.run(['systemctl', 'status', service['service_unit']],
                                 stdout=subprocess.PIPE,
                                 universal_newlines=True,
                                 check=False)
-        journalctl = subprocess.run(['journalctl', '--since', '7 days ago', '-u', service['service_unit']], 
+        journalctl = subprocess.run(['journalctl', '--since', '7 days ago', '-u', service['service_unit'], '--reverse', '--no-hostname'],
                                  stdout=subprocess.PIPE,
                                  universal_newlines=True,
                                  check=False)
@@ -542,7 +693,14 @@ def diagnostic():
         #Replace carrier return to <br> for html view
         sysctl_status = html.escape(sysctl_status.stdout).replace('\n', '<br>')
         journalctl = html.escape(journalctl.stdout).replace('\n', '<br>')
-        active_state = "Active" if service.get('active') == True else "Inactive"
+        try:
+            if service.get('active') == None:
+               service["active"] = ServiceController(service["service_unit"]).isActive()
+            active_state = "Active" if service.get('active') == True else "Inactive"
+        except Exception:
+            active_state =  "Unknown"
+            print(service["service_unit"],"is",active_state)
+
         logs.append({'name' : service['service_unit'], 'active' : active_state, 'sysctl_status' : sysctl_status, 'journalctl' : journalctl})
         
     return render_template('diagnostic.html', logs = logs)
@@ -553,7 +711,7 @@ def get_infos():
     """Small api route to get basic informations about the base station"""
 
     infos = {"app" : "RTKBase",
-             "app_version" : rtkbaseconfig.get("general", "version"), 
+             "app_version" : rtkbaseconfig.get("general", "elt_version"),
              "url" : html.escape(request.base_url),
              "fqdn" : socket.getfqdn(),
              "uptime" : get_uptime(),
@@ -566,11 +724,12 @@ def get_infos():
 def clientConnect():
     global connected_clients
     connected_clients += 1
-    print("Browser client connected")
-    if rtkbaseconfig.get("general", "updated", fallback="False").lower() == "true":
+    updated=rtkbaseconfig.get("general", "updated", fallback="Null")
+    print("Browser client connected. updated =", updated)
+    if updated != "Null":
         rtkbaseconfig.remove_option("general", "updated")
         rtkbaseconfig.write_file()
-        socketio.emit("update_successful", json.dumps({"result": 'true'}), namespace="/test")
+        socketio.emit("update_successful", json.dumps({"result": updated.lower()}), namespace="/test")
     rtk.sendState()
 
 @socketio.on("disconnect", namespace="/test")
@@ -609,6 +768,13 @@ def startBase():
         result = {"result" : "failed"}
         socketio.emit("base starting", json.dumps(result), namespace="/test")
         return
+
+    saved_input_option = ""
+    if saved_input_type == "rtcm3":
+        receiver = rtkbaseconfig.get("main", "receiver").strip("'")
+        if receiver == "u-blox_ZED-X20P":
+           saved_input_option = "-USEB1CINDTEADOFB1I"
+    
     # We must start rtkcv before trying to modify an option
     rtk.startBase()
     saved_input_path = "localhost" + ":" + rtkbaseconfig.get("main", "tcp_port").strip("'")
@@ -618,6 +784,9 @@ def startBase():
     if rtk.get_rtkcv_option("inpstr1-format") != saved_input_type:
         rtk.set_rtkcv_option("inpstr1-format", saved_input_type)
         rtk.set_rtkcv_pending_refresh(True)
+    if rtk.get_rtkcv_option("inpstr1-rcvopt") != saved_input_option:
+        rtk.set_rtkcv_option("inpstr1-rcvopt", saved_input_option)
+        rtk.set_rtkcv_pending_refresh(True)
 
     if rtk.get_rtkcv_pending_status():
         print("REFRESH NEEDED !!!!!!!!!!!!!!!!")
@@ -645,16 +814,18 @@ def deleteLog(json_msg):
 def detect_receiver(json_msg):
     print("Detecting gnss receiver")
     #print("DEBUG json_msg: ", json_msg)
-    answer = subprocess.run([os.path.join(rtkbase_path, "tools", "install.sh"), "--user", rtkbaseconfig.get("general", "user"), "--detect-gnss", "--no-write-port"], encoding="UTF-8", stderr=subprocess.PIPE, stdout=subprocess.PIPE, check=False)
+    answer = subprocess.run([os.path.join(rtkbase_path, "tools", "UnicoreConfigure.sh"), "--user", rtkbaseconfig.get("general", "user"), "--detect-gnss", "--no-write-port"], encoding="UTF-8", stderr=subprocess.PIPE, stdout=subprocess.PIPE, check=False)
     if answer.returncode == 0 and "/dev/" in answer.stdout:
         print(answer.stdout)
         try:
             device_info = next(x for x in answer.stdout.splitlines() if x.startswith('/dev/')).split(' - ')
-            port, gnss_type, speed, firmware, model = [x.strip() for x in device_info]
-            result = {"result" : "success",
+            port, gnss_type, speed, recv_format, firmware, model = [x.strip() for x in device_info]
+            status = "success" if port != "/dev/" else "failed"
+            result = {"result" : status,
                       "port" : port,
                       "gnss_type" : gnss_type,
                       "port_speed" : speed,
+                      "recv_format" : recv_format,
                       "firmware" : firmware,
                       "model" : model}
             result.update(json_msg)
@@ -672,10 +843,19 @@ def detect_receiver(json_msg):
 def apply_receiver_settings(json_msg):
     print("Applying gnss receiver new settings")
     print(json_msg)
-    rtkbaseconfig.update_setting("main", "com_port", json_msg.get("port").strip("/dev/"), write_file=False)
-    rtkbaseconfig.update_setting("main", "com_port_settings", json_msg.get("port_speed") + ':8:n:1', write_file=False)
+    port = json_msg.get("port").strip("/dev/")
+    speed = json_msg.get("port_speed")
+    rtkbaseconfig.update_setting("main", "com_port", port, write_file=False)
+    rtkbaseconfig.update_setting("main", "com_port_settings", speed + ':8:n:1', write_file=False)
     rtkbaseconfig.update_setting("main", "receiver", json_msg.get("gnss_type") + '_' + json_msg.get("model"), write_file=False)
+    rtkbaseconfig.update_setting("main", "receiver_format", json_msg.get("recv_format"), write_file=False)
     rtkbaseconfig.update_setting("main", "receiver_firmware", json_msg.get("firmware"), write_file=True)
+    receiver_conf = os.path.join(rtkbase_path, "receiver.conf")
+    with open(receiver_conf, "w") as f:
+        f.write("recv_port=" + port +"\n")
+        f.write("recv_speed=" + speed + "\n")
+        f.write("recv_position=\n")
+        f.close()
 
     socketio.emit("gnss_settings_saved", json.dumps(json_msg), namespace="/test")
 
@@ -693,8 +873,8 @@ def configure_receiver(brand="", model=""):
         restart_main = False
 
     print("configuring {} gnss receiver model {}".format(brand, model))
-    answer = subprocess.run([os.path.join(rtkbase_path, "tools", "install.sh"), "--user", rtkbaseconfig.get("general", "user"), "--configure-gnss"], encoding="UTF-8", stderr=subprocess.PIPE, stdout=subprocess.PIPE, check=False)
-    print(answer.stdout)
+    answer = subprocess.run([os.path.join(rtkbase_path, "tools", "UnicoreConfigure.sh"), "--user", rtkbaseconfig.get("general", "user"), "--configure-gnss"], encoding="UTF-8", stderr=subprocess.PIPE, stdout=subprocess.PIPE, check=False)
+    #print("DEBUG - stdout: ", answer.stdout)
     #print("DEBUG - returncode: ", answer.returncode)
 
     if answer.returncode == 0: # and "Done" in answer.stdout:
@@ -813,6 +993,11 @@ def rebootRtkbase():
     rtk.shutdown()
     #socketio.stop() hang. I disabled it
     #socketio.stop()
+    answertext = subprocess.check_output([os.path.join(rtkbase_path, "tools", "reboot.sh")], stderr=subprocess.STDOUT, shell=True)
+    if answertext != "":
+       print(answertext.decode('utf-8'))
+    rebooted_state = True
+    sys.stdout.flush()
     subprocess.check_output("reboot")
 
 @socketio.on("shutdown device", namespace="/test")
@@ -823,6 +1008,15 @@ def shutdownRtkbase():
     #socketio.stop()
     subprocess.check_output(["shutdown", "now"])
 
+@socketio.on("update and upgrade", namespace="/test")
+def upgradeSystem():
+    print("upgrade system")
+    answer = subprocess.run([os.path.join(rtkbase_path, "tools", "system_upgrade.sh")], encoding="UTF-8", stderr=subprocess.PIPE, stdout=subprocess.PIPE, check=False)
+    if answer.returncode == 0:
+        socketio.emit("upgrade_finished", "Upgrade succesfull", namespace="/test")
+    else:
+        socketio.emit("upgrade_finished", "Upgrade NOT succesfull", namespace="/test")
+
 @socketio.on("turn off wi-fi", namespace="/test")
 def turnOffWiFi():
     print("Turning off wi-fi")
@@ -881,6 +1075,13 @@ def restartServices(restart_services_list=None):
                     rtk.stopBase()
                     service["unit"].restart()
                 else:
+                    if service.get("flag") == True:
+                        path = os.path.join(rtkbase_path, service.get("name") + ".flg")
+                        if os.path.exists(path):
+                           try:
+                               os.remove(path)
+                           except Exception as e:
+                               print("Remove flag ", path, " error: ", e)
                     service["unit"].restart()
 
     #refresh service status
@@ -904,7 +1105,12 @@ def getServicesStatus(emit_pingback=True):
             #print("unit qui déconne : ", service["name"])
             service["active"] = service["unit"].isActive()
             service["status"] = service["unit"].status()
-            service["result"] = service["unit"].get_result()
+
+            try:
+                service["result"] = service["unit"].get_result()
+            except AttributeError as e:
+                service["result"] = ""
+
             if service.get("result") == "success" and service.get("status") == "running":
                 service["state_ok"] = True
             elif service.get("result") == "exit-code":
@@ -913,7 +1119,10 @@ def getServicesStatus(emit_pingback=True):
                 service["state_ok"] = None
 
         except Exception as e:
-            print("Error getting service info for: {} - {}".format(service['name'], e))
+            if rebooted_state != False:
+                print("Error getting service info for: {} - {}".format(service['name'], e))
+            #TODO manage better the error with rtkbase_archive.service. See https://github.com/Stefal/rtkbase/issues/162
+            #and try to remove this "pass" without any notification (bad practive)
             pass
 
     services_status = []
@@ -984,11 +1193,17 @@ def update_settings(json_msg):
 
         #Restart service if needed
         if source_section == "main":
-            restartServices(("main", "ntrip_A", "ntrip_B", "local_ntrip_caster", "rtcm_svr", "rtcm_client", "rtcm_udp_svr", "rtcm_udp_client", "file", "rtcm_serial", "raw2nmea"))  
+            restartServices(("main", "ntrip_A", "ntrip_B", "ntrip_C", "ntrip_D", "ntrip_E", "local_ntrip_caster", "rtcm_svr", "rtcm_client", "rtcm_udp_svr", "rtcm_udp_client", "file", "rtcm_serial", "raw2nmea"))  
         elif source_section == "ntrip_A":
             restartServices(("ntrip_A",))
         elif source_section == "ntrip_B":
             restartServices(("ntrip_B",))
+        elif source_section == "ntrip_C":
+            restartServices(("ntrip_C",))
+        elif source_section == "ntrip_D":
+            restartServices(("ntrip_D",))
+        elif source_section == "ntrip_E":
+            restartServices(("ntrip_E",))
         elif source_section == "local_ntrip_caster":
             restartServices(("local_ntrip_caster",))
         elif source_section == "rtcm_svr":
@@ -1004,6 +1219,17 @@ def update_settings(json_msg):
         elif source_section == "local_storage":
             restartServices(("file",))
 
+#### Detect GNSS receiver button handler ####
+
+@socketio.on("get tailscale href", namespace="/test")
+def get_tailscale_href():
+    print("get tailscale href")
+    answer = subprocess.run([os.path.join(rtkbase_path, "tools", "tailscale_get_href.sh")], encoding="UTF-8", stderr=subprocess.PIPE, stdout=subprocess.PIPE, check=False)
+    if answer.returncode == 0:
+        socketio.emit("tailscale_href", answer.stdout, namespace="/test")
+    else:
+        socketio.emit("tailscale_error", answer.stdout, namespace="/test")
+
 def arg_parse():
     parser = argparse.ArgumentParser(
         description="RTKBase Web server",
