diff --git a/unit/str2str_ntrip_A.service b/unit/str2str_ntrip_A.service
index 43730b6..3e23716 100644
--- a/unit/str2str_ntrip_A.service
+++ b/unit/str2str_ntrip_A.service
@@ -10,6 +10,7 @@ Type=simple
 SyslogIdentifier=str2str_ntrip_A
 User={user}
 ExecStart={script_path}/run_cast.sh in_tcp out_caster_A
+ExecStopPost={script_path}/tools/onoffELT0x33.sh SETTINGS OFF 0
 Restart=on-failure
 RestartSec=30
 #Limiting log to 1 msg per minute
